pkg/machinepool/autoscaling.go

```diff
@@ -0,0 +1,41 @@
+package machinepool
+
+import (
+	"fmt"
+	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
+	"github.com/giantswarm/azure-admission-controller/pkg/mutator"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	"strings"
+)
+
+// Ensure the needed escapes are in place. See https://tools.ietf.org/html/rfc6901#section-3 .
+func escapeJSONPatchString(input string) string {
+	input = strings.ReplaceAll(input, "~", "~0")
+	input = strings.ReplaceAll(input, "/", "~1")
+
+	return input
+}
+
+// setDefaultSpecValues checks if some optional field is not set, and sets
+// default values defined by upstream Cluster API.
+func ensureAutoscalingAnnotations(m mutator.Mutator, machinePool *capiexp.MachinePool) []mutator.PatchOperation {
+	var patches []mutator.PatchOperation
+
+	// The replicas field could not be set, we default to 1.
+	clusterReplicas := int32(1)
```

Comment:
The default `Spec.Replicas` is set here https://github.com/giantswarm/azure-admission-controller/blob/2740aedcf2dcad316fd8355b77469962b8c91f3b/pkg/machinepool/defaults.go#L28

(but that `Spec.Replicas` value might not be still set in the CR if this is the same webhook request, so this check is also needed)

We should promote a common constant and use it in both places, as it's a default value according to the CAPI spec.

Reply:
done